Proper RFC3339 support

x0b opened the pull request:
Use a RFC 3339 Parser for modtime parsing as specified by
https://github.com/ncw/rclone/blob/e56be0dfd8bf826fc6b2734032f914da420e4431/cmd/lsjson/lsjson.go#L49
Pull request https://github.com/kaczmarkiewiczp/rcloneExplorer/pull/101 does not correctly parse arbitrary timezone offset strings like ```2018-06-05T00:00:00+01:30```


**Full Disclosure**
The parser library is one of my projects, although with 100% test coverage, so this shouldn't be a problem.

buywetwok committed:
Remove duplicated code

buywetwok committed:
Consider timezone when performing last modified time calculations

patrykcoding committed:
Password generator for crypt remote

x0b committed:
proper RFC3339 support

buywetwok committed:
Fix code styling

kaczmarkiewiczp commented:
Thanks, but for now I'm going to go with #101 . I'm a fan of simple solutions that don't require external libraries when possible.

I also found an issue you might be interested about. Some dates are displayed as "Dec 31 1969". I confirmed this behaviour with three different remotes.

I found the problem to be that your library will throw an `Invalid delimiter` exception, and then the `dateInMillis` is assigned a value of 0. Here's an example of a value that gets passed into your library that causes such an exception
`2018-03-27T17:22:26.14Z`

patrykcoding committed:
Merge pull request #101 from buywetwok/topic-last-modified-timestamp-with-timezone

Last modified timestamp calculated incorrectly

patrykcoding committed:
Merge pull request #106 from kaczmarkiewiczp/dev

Fix displayed date

patrykcoding committed:
Fix crash

patrykcoding committed:
Fix unresponsive UI when generating password

patrykcoding committed:
Fix "crash reports" setting not being respected

patrykcoding committed:
Improvements to the displayed modified date

patrykcoding committed:
Don't display modified date for folders

Some remotes don't support modified time and instead show current time.
Hide the modified date for those remotes

patrykcoding committed:
Fix crash when loading remotes

patrykcoding committed:
Merge pull request #107 from kaczmarkiewiczp/dev

Password generator when creating crypt remote

patrykcoding committed:
Fix crash with thumbnail service

patrykcoding committed:
Merge pull request #108 from kaczmarkiewiczp/dev

Fix crash with thumbnail service

x0b committed:
time zone support for modtimes

x0b committed:
reverse mere

x0b committed:
fixed off-by-one-error

x0b commented:
Dammit. Thank you for trying anyways. 

However, considering that non-utc timestamps are still not supported I've reimplemented the timezone parsing and it seems to work so far.
Tested with
- box.com
- dropbox.com
- hubic
- ondrive
- webdav
- crypt remotes

The commit is based on top of #102 to allow conflict-free merging

kaczmarkiewiczp commented on app/src/main/java/ca/pkay/rcloneexplorer/Items/FileItem.java:
```diff
@@ -94,12 +94,22 @@ private String sizeToHumanReadable(long size) {
     }
 
     private long modTimeToMilis(String modTime) {
+        if(modTime.lastIndexOf("+") > 18 || modTime.lastIndexOf("-") > 18){
```
Needs a space between the `if` and opening brace, as well as the closing brace and curly bracket

kaczmarkiewiczp commented on app/src/main/java/ca/pkay/rcloneexplorer/Items/FileItem.java:
```diff
@@ -111,23 +121,27 @@ private long modTimeToMilis(String modTime) {
-        long dateInMillis;
-        Date date;
-        SimpleDateFormat simpleDateFormat = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss");
+    private long modTimeZonedToMillis(String modTime){
```
Also space before the opening curly brace

kaczmarkiewiczp commented on app/src/main/java/ca/pkay/rcloneexplorer/Items/FileItem.java:
```diff
@@ -111,23 +121,27 @@ private long modTimeToMilis(String modTime) {
+    private long modTimeZonedToMillis(String modTime){
+        SimpleDateFormat format = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ssXXX");
+        int index = modTime.lastIndexOf("+");
+        if(index == -1){
```
A space between the `if` and opening brace, and before the opening curly brace.

patrykcoding reviewed (changes requested):
Looks like the code is working great now, thanks. There are some styling problems concerning [horizontal whitespace](https://google.github.io/styleguide/javaguide.html#s4.6.2-horizontal-whitespace). I left inline notes. If you could please address that then I'll merge it in. Thanks.

x0b committed:
fixed whitespace
